x-pack/test/detection_engine_api_integration/basic/tests/patch_rules_bulk.ts

```diff
@@ -45,6 +45,7 @@ export default ({ getService }: FtrProviderContext) => {
         const { body } = await supertest
           .patch(DETECTION_ENGINE_RULES_BULK_UPDATE)
           .set('kbn-xsrf', 'true')
+          .set('elastic-api-version', '2023-10-31')
```

Comment:
So after about the 100th of these (and I'm sure you asked yourself the same thing while plumbing them through 😅), but can we not set a base version on via a `supertest` config and then override from there? Or is this a bad idea and we should always explicitly set them?

Reply:
> One approach might be to lean on code generation to create an API client specifically for tests. 

I really like this idea! Thanks for the details @xcrzx! 🙂